crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs

```diff
@@ -723,6 +731,100 @@ impl KitsuneP2pHandler for Space {
+                            ro_inner.clone(),
+                            agent.clone(),
+                            timeout,
+                        )
+                        .await;
+                        (result, agent)
+                    };
+                    futures.push(discover_fut);
+                }
+                let futures = futures::stream::iter(futures);
+                let futures = futures.buffer_unordered(concurrent_limit_per_thread);
+                futures
+                    .for_each_concurrent(concurrent_limit_per_thread, |(discover_result, agent)| {
+                        match discover_result {
+                            discover::PeerDiscoverResult::OkShortcut => {
+                                // reflect this request locally
+                                evt_sender
+                                    .notify(
+                                        space.clone(),
+                                        agent,
+                                        from_agent.clone(),
+                                        payload.clone(),
+                                    )
+                                    .map(|r| {
+                                        if let Err(e) = r {
+                                            tracing::error!(
+                                                "Failed to broadcast to local agent because: {:?}",
+                                                e
+                                            )
+                                        }
+                                    })
+                                    .boxed()
+                            }
+                            discover::PeerDiscoverResult::OkRemote { con_hnd, .. } => {
+                                let payload = wire::Wire::broadcast(
+                                    space.clone(),
+                                    agent,
+                                    payload.clone().into(),
+                                );
+                                con_hnd
+                                    .notify(&payload, timeout)
+                                    .map(|r| {
+                                        if let Err(e) = r {
+                                            tracing::info!(
+                                                "Failed to broadcast to remote agent because: {:?}",
+                                                e
+                                            )
+                                        }
+                                    })
+                                    .boxed()
+                            }
+                            discover::PeerDiscoverResult::Err(e) => async move {
+                                tracing::info!(
+                                    "Failed to discover connection for {:?} because: {:?}",
+                                    agent,
+                                    e
+                                );
```

Comment:
Fire-n-forget errors elsewhere in kitsune use `warn!` level... but now that I'm thinking about it, perhaps we should downgrade them all to `debug!` if we're expecting the "n-forget" part to be pretty common, maybe we don't want them showing up in our logs much?

Reply:
Yeh debug could be a good idea. With this broadcast it's very likely to have failures because it's taking an arbitrary list of agents.
I mean the failure to discover is the most likely. You could say once discovered that failure to notify should be less likely but either way we probably don't want it in the logs unless we are specifically concerned with lots of abnormal broadcast failures.